src/preact/component/renderer.js

```diff
@@ -0,0 +1,51 @@
+/**
```

Comment:
Is this the correct directory for this?

Reply:
The idea of this dir is all the shared code (components and hooks) that our components can use. Maybe not a very good name for this dir. What I meant was more like "for components" or "component support". We basically end up with three major areas:

1. Preact declarations. This is just `src/preact/index.js` and `src/preact/compat.js`
2. Preact/AMP bridge. Includes things like `PreactBaseElement`, `Slot`, etc - all the stuff that an `amp-element.js` would use.
3. Shared component support. Includes wrappers, hooks, etc - all the stuff that a "pure" component in a `component.js` would use.

So I started reorganizing it in this vein, but not finished yet. Naming suggestions are welcome.